tests/framework/cli/test_starters.py

```diff
@@ -321,14 +320,25 @@ def test_parse_tools_invalid_range(self, input, capsys):
         message = f"'{input}' is an invalid range for project tools.\nPlease ensure range values go from smaller to larger."
         assert message in capsys.readouterr().err
 
+    @pytest.mark.parametrize(
+        "input,right_border",
+        [("3-9", "9"), ("3-10000", "10000")],
+    )
+    def test_parse_tools_range_too_high(self, input, right_border, capsys):
+        with pytest.raises(SystemExit):
+            selected = _parse_tools_input(input)
+            _validate_tool_selection(selected)
```

Comment:
I think it will protect us from "0-5"

Reply:
I'm not sure that should be a test case for this test as we're checking that the error is thrown when parsing a range that extends beyond available tools. We'd want to ensure this error is being correctly raised by `_parse_tools_input` instead of later in `_validate_tool_selection `.

Reply:
yes, we can make a full validation inside of _parse_tools_input() and remove _validate_tool_selection(selected), I agree

Reply:
Apologies, I misunderstood and thought you were suggesting to completely remove the _validate_tool_selection(selected) function. I've addressed your comment now.